core/src/cluster_slot_state_verifier.rs

```diff
@@ -666,6 +676,14 @@ fn on_popular_pruned_fork(slot: Slot) -> Vec<ResultingStateChange> {
     )]
 }
 
+/// Finds the cluster confirmed hash
+///
+/// 1) If we have a frozen hash, check if it's been duplicate confirmed by cluster
+///    in turbine or gossip
+/// 2) Otherwise poll `epoch_slots_frozen_slots` to see if we have a hash
+///
+/// Note `epoch_slots_frozen_slots` is not populated from `EpochSlots` in gossip but actually
```

Comment:
also for a follow up pr, maybe we can rename this because it doesn't contain all epoch frozen slots but only slots which we have hashes for as received by ancestor_hashes_service.

Reply:
yep exactly! i like the name `ancestor_hashes_results`

Reply:
tracking here https://github.com/solana-labs/solana/issues/32287